src/main/java/ui/components/FilterTextField.java

```diff
@@ -94,6 +97,18 @@ private void setup() {
     }
 
     /**
+     * Set the background of FilterTextField to green if there is no syntax errors.
```

Comment:
I suggested not to use green because the only case that needs to be highlighted is errors. If there is no error, we don't have to do anything special.